lapis2/src/main/kotlin/org/genspectrum/lapis/request/LapisInfo.kt

```diff
@@ -29,6 +39,15 @@ class ResponseBodyAdviceDataVersion(private val dataVersion: DataVersion) : Resp
         response: ServerHttpResponse,
     ): Any? {
         response.headers.add(LAPIS_DATA_VERSION_HEADER, dataVersion.dataVersion)
+
+        if (body is LapisResponse<*>) {
+            return LapisResponse(body.data, LapisInfo(dataVersion.dataVersion))
+        }
+
+        if (body is LapisErrorResponse) {
+            return LapisErrorResponse(body.error, LapisInfo(dataVersion.dataVersion))
+        }
```

Comment:
```suggestion
        when (body) {
            is LapisResponse<*> -> body.info.dataVersion = dataVersion.dataVersion
            is LapisErrorResponse -> body.info.dataVersion = dataVersion.dataVersion
        }
```
or else make `LapisInfo.dataVersion` a `val`